src/arr/trove/lists.arr

```diff
@@ -887,6 +887,30 @@ fun distinct(l :: List) -> List:
   end
 end
 
+fun take-while<A>(pred :: (A -> Boolean), lst :: List<A>) -> {List<A>; List<A>}:
+  doc: "Splits a list into two pieces, at the first element that fails the given predicate"
+  var tail = empty
+  fun help(l):
+    cases(List) l:
+      | empty => empty
+      | link(first, rest) =>
+        if pred(first) block:
+          link(first, help(rest))
+        else:
+          tail := l
+          empty
+        end
+    end
+  end
+  { help(lst); tail }
+where:
```

Comment:
Separate issue.  I was following the exemplars in the existing lists file; we should clean it all up at once.